tests/python/states/slater_test.py

```diff
@@ -0,0 +1,90 @@
+"""Test Slater sampler."""
+
+from __future__ import annotations
+
+import itertools
+
+import numpy as np
+import pytest
+
+import ffsim
+from ffsim.states.bitstring import BitstringType
+
+
+@pytest.mark.parametrize(
+    "norb, nelec, bitstring_type",
+    [
+        (norb, nelec, bitstring_type)
+        for (norb, nelec), bitstring_type in itertools.product(
+            ffsim.testing.generate_norb_nelec(range(1, 5)), BitstringType
+        )
+    ],
+)
+def test_slater_sampler(
+    norb: int, nelec: tuple[int, int], bitstring_type: BitstringType
+):
+    """Test Slater determinant sampler."""
+
+    rng = np.random.default_rng(1234)
+    shots = 3000
+    n_a, n_b = nelec
+
+    rotation_a = ffsim.random.random_unitary(norb, seed=rng)
+    rotation_b = ffsim.random.random_unitary(norb, seed=rng)
+    rdm_a, rdm_b = ffsim.slater_determinant_rdms(
+        norb, (range(n_a), range(n_b)), (rotation_a, rotation_b)
```

Comment:
Instead of only testing the occupied orbitals being `(range(n_a), range(n_b))`, let's iterate over all the possibilities. This is more thorough and will cover edge cases. For an example of this, see:
https://github.com/qiskit-community/ffsim/blob/57bb49f7cd4a52b5c36cacea9d96b7c37d25c034/tests/python/gates/diag_coulomb_test.py#L159-L162

Same comment applies for the spinless test.

Reply:
Just added it! The tests are a bit slow at the moment

Reply:
How long do they take to run?